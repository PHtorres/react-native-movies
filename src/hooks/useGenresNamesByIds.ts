import { useGenres } from '../state/hooks/useGenres';

export function useGenresNamesByIds(ids: number[]) {
  const { genres } = useGenres();
  const genresNames = genres
    .filter(genre => ids.includes(genre.id))
    .map(genre => genre.name)
    .join(', ');
  return { genresNames };
}
