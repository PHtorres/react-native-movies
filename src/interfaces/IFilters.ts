export interface IFilters {
  with_genres?: string;
  sort_by?: string;
  year?: number;
  ["with_runtime.gte"]?: number;
  ["with_runtime.lte"]?: number;
}
