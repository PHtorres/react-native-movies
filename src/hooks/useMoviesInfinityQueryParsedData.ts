import { useMemo } from 'react';
import { AxiosResponse } from 'axios';
import { IMovies } from '../interfaces/IMovies';
import { IMovie } from '../interfaces/IMovie';

export function useMoviesInfinityQueryParsedData(pages: AxiosResponse<IMovies>[] | undefined) {
  const parsedData = useMemo(() => {
    let resultList: IMovie[] = [];
    pages?.forEach(page => {
      resultList = resultList.concat(page.data!.results);
    });
    return resultList;
  }, [pages]);

  return {
    parsedData,
  };
}
