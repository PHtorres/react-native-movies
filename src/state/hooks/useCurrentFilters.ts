import { useMemo } from 'react';
import { create } from 'zustand';
import { IFilters } from '../../interfaces/IFilters';
import { useGenresOptions } from './useGenresOptions';
import { useSortOption } from './useSortOption';
import { objectAsString } from '../../utils/object';
import { useYearOption } from './useYearOption';
import { useRuntimeOption } from './useRuntimeOption';

interface FiltersActions {
  filters: IFilters;
  setFilters(filters: IFilters): void;
}

export const useFilters = create<FiltersActions>(set => ({
  filters: {},
  setFilters: value => {
    set(() => ({ filters: value }));
  },
}));

export function useCurrentFilters() {
  const { filters, setFilters } = useFilters();
  const { sortOption } = useSortOption();
  const { selectedGenresIds } = useGenresOptions();
  const { selectedYear } = useYearOption();
  const { runtimeGte, runtimeLte } = useRuntimeOption();

  function updateCurrentFilters() {
    setFilters({
      sort_by: sortOption,
      with_genres: selectedGenresIds.join(', '),
      year: selectedYear,
      'with_runtime.gte': runtimeGte,
      'with_runtime.lte': runtimeLte,
    });
  }

  const filtersKey = useMemo(() => {
    return objectAsString(filters);
  }, [filters]);

  return {
    filters,
    filtersKey,
    updateCurrentFilters,
  };
}
