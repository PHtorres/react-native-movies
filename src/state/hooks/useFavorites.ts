import AsyncStorage from '@react-native-async-storage/async-storage';
import { create } from 'zustand';
import { persist, createJSONStorage } from 'zustand/middleware';
import { IMovie } from '../../interfaces/IMovie';

interface IFavoritesActions {
  favorites: IMovie[];
  toggleFavorite: (favorite: IMovie) => void;
  isFavorite: (id: number) => boolean;
}

function removeFavorite(favorites: IMovie[], favorite: IMovie) {
  const newFavorites = favorites.filter(f => f.id !== favorite.id);
  return newFavorites;
}

function addFavorite(favorites: IMovie[], favorite: IMovie) {
  favorites.unshift(favorite);
  return favorites;
}

function toggleFavorite(favorites: IMovie[], favorite: IMovie) {
  if(isFavorite(favorites, favorite.id)){
    return removeFavorite(favorites, favorite);
  }
  return addFavorite(favorites, favorite);
}

function isFavorite(favorites: IMovie[], id: number) {
  const result = favorites.find(f => f.id == id);
  return !!result;
}

export const useFavorites = create<IFavoritesActions>()(
  persist(
    (set, props) => ({
      favorites: [],
      toggleFavorite: favorite => set({ favorites: toggleFavorite(props().favorites, favorite) }),
      isFavorite: id => isFavorite(props().favorites, id),
    }),
    {
      name: 'MoviesOnTheGo@Favorites',
      storage: createJSONStorage(() => AsyncStorage),
    },
  ),
);
