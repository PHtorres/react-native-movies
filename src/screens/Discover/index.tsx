import { View, ActivityIndicator } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { FlashList } from '@shopify/flash-list';
import { useInfiniteQuery } from '@tanstack/react-query';
import EmptyState from '../../components/EmptyState';
import { colors, screenStyles } from '../../shared/styles';
import { MovieCardLandscape } from '../../components/MovieCardLandscape';
import { requestsKeys } from '../../api/requests/shared/keys';
import { discoverRequests } from '../../api/requests/discover';
import { useMoviesInfinityQueryParsedData } from '../../hooks/useMoviesInfinityQueryParsedData';
import { MOVIE_CARD_HEIGHT, MOVIE_CARD_MARGIN } from '../../components/MovieCardLandscape/styles';
import { useCurrentFilters } from '../../state/hooks/useCurrentFilters';
import { elementsIds } from '../../tests/elementsIds';

export function Discover() {
  const { navigate } = useNavigation();
  const { filters, filtersKey } = useCurrentFilters();

  const { data, fetchNextPage, hasNextPage, isFetchingNextPage } = useInfiniteQuery({
    keepPreviousData: true,
    queryKey: requestsKeys.discover_movie(filtersKey),
    queryFn: ({ pageParam = 1 }) => discoverRequests.getMovies(pageParam, filters),
    getNextPageParam: (lastPage, _) => {
      return lastPage.data?.page < lastPage.data?.total_pages ? lastPage.data?.page + 1 : undefined;
    },
  });

  const { parsedData } = useMoviesInfinityQueryParsedData(data?.pages);

  const loadMore = async () => {
    hasNextPage && (await fetchNextPage());
  };

  return (
    <View style={screenStyles.container}>
      <FlashList
        testID={elementsIds.discoverList}
        data={parsedData}
        renderItem={({ item }) => <MovieCardLandscape movie={item} />}
        estimatedItemSize={MOVIE_CARD_HEIGHT}
        alwaysBounceVertical={true}
        ItemSeparatorComponent={() => <View style={{ height: MOVIE_CARD_MARGIN }} />}
        ListEmptyComponent={() => (
          <EmptyState
            image={require('../../assets/empty-discover.jpg')}
            title="No results found"
            message="Try adjusting the settings"
            actionLabel="Go to Settings"
            onAction={() => navigate('Settings')}
          />
        )}
        onEndReached={loadMore}
        onEndReachedThreshold={0.1}
        ListFooterComponent={() =>
          isFetchingNextPage ? <ActivityIndicator color={colors.primary} size="large" /> : null
        }
      />
    </View>
  );
}
