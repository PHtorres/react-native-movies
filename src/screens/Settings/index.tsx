import {
  KeyboardAvoidingView,
  ScrollView,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import Icon from 'react-native-vector-icons/Ionicons';

import { colors, screenStyles, sizes, textStyles } from '../../shared/styles';
import { styles } from './style';
import { useSortOption } from '../../state/hooks/useSortOption';
import { useGenresOptions } from '../../state/hooks/useGenresOptions';
import { SORT_OPTIONS } from '../../constants/sortOptions';
import { useGenres } from '../../state/hooks/useGenres';
import { useCurrentFilters } from '../../state/hooks/useCurrentFilters';
import { useYearOption } from '../../state/hooks/useYearOption';
import { useRuntimeOption } from '../../state/hooks/useRuntimeOption';
import { memo } from 'react';
import { IS_IOS } from '../../utils/constants';
import { elementsIds } from '../../tests/elementsIds';

/**
 * To ensure that the keyboard will not be displayed over the form area
 */
const KEYBOARD_VERTICAL_OFFSET = 50;

export function Settings() {
  const { goBack } = useNavigation();
  const insets = useSafeAreaInsets();
  const { updateCurrentFilters } = useCurrentFilters();

  function handleSave() {
    updateCurrentFilters();
    goBack();
  }

  return (
    <KeyboardAvoidingView
      behavior={IS_IOS ? 'padding' : 'height'}
      style={{ flex: 1 }}
      keyboardVerticalOffset={KEYBOARD_VERTICAL_OFFSET}
    >
      <View
        style={[
          screenStyles.container,
          {
            paddingBottom: insets.bottom,
          },
        ]}
      >
        <ScrollView contentContainerStyle={styles.wrapper}>
          <SortBy />
          <Genres />
          <Year />
          <Runtime />
        </ScrollView>
        <View style={styles.buttonContainer}>
          <TouchableOpacity activeOpacity={0.7} style={styles.button} onPress={handleSave}>
            <Text style={styles.buttonText}>Save</Text>
          </TouchableOpacity>
        </View>
      </View>
    </KeyboardAvoidingView>
  );
}

const SortBy = memo(() => {
  const { sortOption, setSortOption } = useSortOption();
  return (
    <View>
      <Text style={textStyles.h2}>Sort by</Text>
      <View>
        {SORT_OPTIONS.map(option => (
          <SortOption
            key={option.value}
            name={option.label}
            selected={sortOption === option.value}
            onSelect={() => setSortOption(option.value)}
          />
        ))}
      </View>
    </View>
  );
});

const Genres = memo(() => {
  const { genres } = useGenres();
  const { isGenreSelected, toggleSelectedGenre } = useGenresOptions();
  return (
    <View>
      <Text style={textStyles.h2}>Genres</Text>
      <View style={styles.genreList}>
        {genres.map(genre => (
          <Genre
            key={genre.id}
            name={genre.name}
            selected={isGenreSelected(genre.id)}
            onSelect={() => toggleSelectedGenre(genre.id)}
          />
        ))}
      </View>
    </View>
  );
});

const Year = memo(() => {
  const { selectedYear, setSelectedYear } = useYearOption();
  return (
    <View>
      <Text style={textStyles.h2}>Year</Text>
      <TextInput
        testID={elementsIds.yearInput}
        keyboardType="number-pad"
        style={styles.input}
        maxLength={4}
        value={selectedYear?.toString()}
        onChangeText={value => setSelectedYear(value ? Number(value) : undefined)}
        returnKeyType="done"
      />
    </View>
  );
});

const Runtime = memo(() => {
  const { runtimeGte, runtimeLte, setSelectedRuntimeGte, setSelectedRuntimeLte } =
    useRuntimeOption();
  return (
    <View>
      <Text style={textStyles.h2}>Runtime</Text>
      <View style={styles.runtime}>
        <TextInput
          testID={elementsIds.runtimeFromInput}
          keyboardType="number-pad"
          style={styles.input}
          placeholder="From"
          placeholderTextColor={colors.neutral}
          maxLength={3}
          value={runtimeGte?.toString()}
          onChangeText={value => setSelectedRuntimeGte(value ? Number(value) : undefined)}
          returnKeyType="done"
        />
        <Text style={textStyles.small}>-</Text>
        <TextInput
          testID={elementsIds.runtimeToInput}
          keyboardType="number-pad"
          style={styles.input}
          placeholder="To"
          placeholderTextColor={colors.neutral}
          maxLength={3}
          value={runtimeLte?.toString()}
          onChangeText={value => setSelectedRuntimeLte(value ? Number(value) : undefined)}
          returnKeyType="done"
        />
        <Text style={textStyles.small}>minutes</Text>
      </View>
    </View>
  );
});

interface OptionProps {
  name: string;
  selected: boolean;
  onSelect: () => void;
}

function Genre({ name, selected, onSelect }: OptionProps) {
  return (
    <TouchableOpacity
      testID={`genre-option-${name}-${selected ? 'selected' : ''}`}
      style={[styles.genre, selected ? styles.selectedGenre : undefined]}
      activeOpacity={0.7}
      onPress={onSelect}
    >
      <Text style={[selected ? styles.selectedGenreText : styles.optionLabel]}>{name}</Text>
      {selected && (
        <Icon
          testID={`genre-option-icon-${name}-selected`}
          name="close-outline"
          size={sizes.lg}
          color={colors.white}
        />
      )}
    </TouchableOpacity>
  );
}

function SortOption({ name, selected, onSelect }: OptionProps) {
  return (
    <TouchableOpacity
      testID={`sort-option-${name}-${selected ? 'selected' : ''}`}
      style={styles.sortOption}
      activeOpacity={0.7}
      onPress={onSelect}
    >
      <Text style={styles.optionLabel}>{name}</Text>
      <Icon
        testID={`sort-option-icon-${name}-${selected ? 'selected' : ''}`}
        name={selected ? 'checkmark-circle' : 'ellipse-outline'}
        size={sizes.xxl}
        color={selected ? colors.primary : colors.black}
      />
    </TouchableOpacity>
  );
}
