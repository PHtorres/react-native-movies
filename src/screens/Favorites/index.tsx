import { useNavigation } from '@react-navigation/native';
import { useCallback } from 'react';
import { FlatList, ListRenderItemInfo, View } from 'react-native';
import EmptyState from '../../components/EmptyState';
import { MovieCardPortrait } from '../../components/MovieCardPortrait';
import {
  MOVIE_CARD_MARGIN_BOTTOM,
  MOVIE_CARD_HEIGHT,
} from '../../components/MovieCardPortrait/styles';
import { IMovie } from '../../interfaces/IMovie';
import { screenStyles } from '../../shared/styles';
import { useFavorites } from '../../state/hooks/useFavorites';
import { elementsIds } from '../../tests/elementsIds';

export function Favorites() {
  const { navigate } = useNavigation();
  const { favorites } = useFavorites();

  const renderFavoriteItem = useCallback(({ item }: ListRenderItemInfo<IMovie>) => {
    return <MovieCardPortrait movie={item} />;
  }, []);

  return (
    <View style={screenStyles.secondaryContainer}>
      <FlatList
        testID={elementsIds.favoriteList}
        data={favorites}
        numColumns={2}
        keyExtractor={item => item.id.toString()}
        renderItem={renderFavoriteItem}
        ItemSeparatorComponent={() => <View style={{ height: MOVIE_CARD_MARGIN_BOTTOM }} />}
        ListEmptyComponent={() => (
          <EmptyState
            image={require('../../assets/empty-favorites.jpg')}
            title="You haven't liked any movie yet"
            message="Why not try to find a movie you like?"
            actionLabel="Go to Discover"
            onAction={() => navigate('Discover')}
          />
        )}
        //settings to improve performance
        removeClippedSubviews={true}
        maxToRenderPerBatch={1}
        initialNumToRender={6}
        getItemLayout={(_, index) => ({
          length: MOVIE_CARD_HEIGHT,
          offset: (MOVIE_CARD_HEIGHT + MOVIE_CARD_MARGIN_BOTTOM) * index,
          index,
        })}
      />
    </View>
  );
}
