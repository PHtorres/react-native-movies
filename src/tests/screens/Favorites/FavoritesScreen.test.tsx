import React from 'react';
import { render, fireEvent, act } from '@testing-library/react-native';
import { renderHook } from '@testing-library/react-hooks';
import waait from 'waait';
import { Favorites } from '../../../screens/Favorites';
import { TestProvider } from '../TestProvider';
import { successMoviesDataPage1 } from '../../mocks/movies';
import { elementsIds } from '../../elementsIds';
import { IMovie } from '../../../interfaces/IMovie';
import { useFavorites } from '../../../state/hooks/useFavorites';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => {
    return { navigate: mockedNavigate };
  },
}));

const renderFavoritesScreen = () => {
  return render(<Favorites />, {
    wrapper: TestProvider,
  });
};

const mockedMovie1 = successMoviesDataPage1.results[0];
const mockedMovie2 = successMoviesDataPage1.results[1];
const mockedMovie3 = successMoviesDataPage1.results[3];

const mockedFavoriteList = [mockedMovie1, mockedMovie2, mockedMovie3];

describe('FavoritesScreen', () => {
  beforeAll(() => {
    const { result } = renderHook(() => useFavorites());
    result.current.toggleFavorite(mockedMovie3);
    result.current.toggleFavorite(mockedMovie2);
    result.current.toggleFavorite(mockedMovie1);
  });
  it('should render without crashing', async () => {
    const result = renderFavoritesScreen();
    await waait(2000);
    expect(result).toMatchSnapshot();
  });

  it('should render favorite movies list', async () => {
    const { getByTestId } = renderFavoritesScreen();
    await waait(2000);
    const listData = getByTestId(elementsIds.favoriteList).props.data as IMovie[];
    expect(listData).toEqual(mockedFavoriteList);
  });

  it('should be able to remove a movie from favorite list', async () => {
    const { getByTestId, queryByText } = renderFavoritesScreen();
    await waait(2000);
    const { result } = renderHook(() => useFavorites());
    expect(result.current.favorites).toContain(mockedMovie1);
    const movieFavoriteButton = getByTestId(mockedMovie1.id.toString());
    await act(async () => {
      fireEvent.press(movieFavoriteButton);
    });
    const itemTitle = queryByText(mockedMovie1.original_title);
    expect(itemTitle).toBeFalsy();
    expect(result.current.favorites).not.toContain(mockedMovie1);

    //add again
    result.current.toggleFavorite(mockedMovie1);
    expect(result.current.favorites).toContain(mockedMovie1);
    const addedItemTitle = queryByText(mockedMovie1.original_title);
    expect(addedItemTitle).toBeTruthy();
  });

  it('should be able to navigate to Discover screen when there is no favorites', async () => {
    const { getByText } = renderFavoritesScreen();
    const { result } = renderHook(() => useFavorites());
    result.current.toggleFavorite(mockedMovie3);
    result.current.toggleFavorite(mockedMovie2);
    result.current.toggleFavorite(mockedMovie1);

    await waait(2000);
    const discoverButton = getByText('Go to Discover');
    await act(async () => {
      fireEvent.press(discoverButton);
    });

    expect(mockedNavigate).toBeCalledWith('Discover');
  });
});
