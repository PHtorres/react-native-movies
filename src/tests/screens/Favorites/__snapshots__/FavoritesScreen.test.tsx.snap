// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesScreen should render without crashing 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    data={
      [
        {
          "adult": false,
          "backdrop_path": "/a2tys4sD7xzVaogPntGsT1ypVoT.jpg",
          "genre_ids": [
            53,
            35,
            80,
            27,
          ],
          "id": 804150,
          "original_language": "en",
          "original_title": "Cocaine Bear",
          "overview": "Inspired by a true story, an oddball group of cops, criminals, tourists and teens converge in a Georgia forest where a 500-pound black bear goes on a murderous rampage after unintentionally ingesting cocaine.",
          "popularity": 2946.378,
          "poster_path": "/gOnmaxHo0412UVr1QM5Nekv1xPi.jpg",
          "release_date": "2023-02-22",
          "title": "Cocaine Bear",
          "video": false,
          "vote_average": 6.6,
          "vote_count": 518,
        },
        {
          "adult": false,
          "backdrop_path": "/kmzppWh7ljL6K9fXW72bPN3gKwu.jpg",
          "genre_ids": [
            28,
            35,
            14,
            80,
          ],
          "id": 1013860,
          "original_language": "en",
          "original_title": "R.I.P.D. 2: Rise of the Damned",
          "overview": "When Sheriff Roy Pulsipher finds himself in the afterlife, he joins a special police force and returns to Earth to save humanity from the undead.",
          "popularity": 315.119,
          "poster_path": "/g4yJTzMtOBUTAR2Qnmj8TYIcFVq.jpg",
          "release_date": "2022-11-15",
          "title": "R.I.P.D. 2: Rise of the Damned",
          "video": false,
          "vote_average": 6.6,
          "vote_count": 440,
        },
        {
          "adult": false,
          "backdrop_path": "/dKqa850uvbNSCaQCV4Im1XlzEtQ.jpg",
          "genre_ids": [
            35,
            80,
            9648,
          ],
          "id": 661374,
          "original_language": "en",
          "original_title": "Glass Onion: A Knives Out Mystery",
          "overview": "World-famous detective Benoit Blanc heads to Greece to peel back the layers of a mystery surrounding a tech billionaire and his eclectic crew of friends.",
          "popularity": 222.228,
          "poster_path": "/vDGr1YdrlfbU9wxTOdpf3zChmv9.jpg",
          "release_date": "2022-11-23",
          "title": "Glass Onion: A Knives Out Mystery",
          "video": false,
          "vote_average": 7.1,
          "vote_count": 3829,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    getItemLayout={[Function]}
    initialNumToRender={6}
    keyExtractor={[Function]}
    maxToRenderPerBatch={1}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="favorite-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "flex-start",
                "gap": 10,
                "height": 634.5,
                "justifyContent": "flex-start",
                "marginHorizontal": 11.25,
                "width": 352.5,
              }
            }
          >
            <View
              style={
                {
                  "height": 528.75,
                }
              }
            >
              <View
                style={{}}
              >
                <View
                  style={
                    {
                      "position": "absolute",
                      "right": 5,
                      "top": 5,
                      "zIndex": 999,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderRadius": 12.5,
                        "height": 25,
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": 25,
                      }
                    }
                    testID="804150"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#0d9488",
                            "fontSize": 16,
                          },
                          undefined,
                          {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      ïŒ®
                    </Text>
                  </View>
                </View>
                <Image
                  source={
                    {
                      "uri": "https://image.tmdb.org/t/p/w200/gOnmaxHo0412UVr1QM5Nekv1xPi.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": "100%",
                      "width": 352.5,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              Cocaine Bear
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "flex-start",
                "gap": 10,
                "height": 634.5,
                "justifyContent": "flex-start",
                "marginHorizontal": 11.25,
                "width": 352.5,
              }
            }
          >
            <View
              style={
                {
                  "height": 528.75,
                }
              }
            >
              <View
                style={{}}
              >
                <View
                  style={
                    {
                      "position": "absolute",
                      "right": 5,
                      "top": 5,
                      "zIndex": 999,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderRadius": 12.5,
                        "height": 25,
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": 25,
                      }
                    }
                    testID="1013860"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#0d9488",
                            "fontSize": 16,
                          },
                          undefined,
                          {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      ïŒ®
                    </Text>
                  </View>
                </View>
                <Image
                  source={
                    {
                      "uri": "https://image.tmdb.org/t/p/w200/g4yJTzMtOBUTAR2Qnmj8TYIcFVq.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": "100%",
                      "width": 352.5,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              R.I.P.D. 2: Rise of the Damned
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "flex-start",
                "gap": 10,
                "height": 634.5,
                "justifyContent": "flex-start",
                "marginHorizontal": 11.25,
                "width": 352.5,
              }
            }
          >
            <View
              style={
                {
                  "height": 528.75,
                }
              }
            >
              <View
                style={{}}
              >
                <View
                  style={
                    {
                      "position": "absolute",
                      "right": 5,
                      "top": 5,
                      "zIndex": 999,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderRadius": 12.5,
                        "height": 25,
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": 25,
                      }
                    }
                    testID="661374"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#0d9488",
                            "fontSize": 16,
                          },
                          undefined,
                          {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      ïŒ®
                    </Text>
                  </View>
                </View>
                <Image
                  source={
                    {
                      "uri": "https://image.tmdb.org/t/p/w200/vDGr1YdrlfbU9wxTOdpf3zChmv9.jpg",
                    }
                  }
                  style={
                    {
                      "borderRadius": 10,
                      "height": "100%",
                      "width": 352.5,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#000000",
                  "fontSize": 14,
                  "fontWeight": "400",
                }
              }
            >
              Glass Onion: A Knives Out Mystery
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
